bigquery/google/cloud/bigquery/client.py

```diff
@@ -2060,6 +2131,12 @@ def query(
 
         if self._default_query_job_config:
             if job_config:
+                if not isinstance(job_config, google.cloud.bigquery.job.QueryJobConfig):
+                    raise ValueError(
+                        "Expected an instance of QueryJobConfig class for the job_config parameter, but received job_config = {}".format(
+                            job_config
+                        )
+                    )
```

Comment:
This check and the error message are repeated at quite a few places in the file, and it would be beneficial to extract it to a helper function, making the code more maintainable.

Something like the following:

```py
def _verify_job_config_type(job_config, expected_type, param_name="job_config", allow_none=True):
    if job_config is None and allow_none:
        return

    if not isinstance(job_config, expected_type):
        msg = (
            "Expected an instance of {expected_type} class for the {param_name} parameter, "
            "but received {param_name} = {job_config}"
        )
        raise TypeError(
            msg.format(
                 expected_type=expected_type.__class__.__name__,
                 param_name=param_name,
                 job_config=job_config,
             )
        )
```

Also, raising a `TypeError` is more appropriate than `ValueError`, as the latter's description is _"Inappropriate argument value (of correct type)."_.